modules/actions/join.js

```diff
@@ -40,12 +39,14 @@ export function actionJoin(ids) {
         });
 
         // Prefer to keep an existing way.
-        for (var i = 0; i < ways.length; i++) {
-            if (!ways[i].isNew()) {
-                survivorID = ways[i].id;
-                break;
-            }
-        }
+        // if there are mulitple existing ways, keep the oldest one
+        const survivorID = (
+            ways
+                .filter((way) => !way.isNew())
+                .sort((a, b) => +new Date(a.timestamp) - +new Date(b.timestamp))[0] || ways[0]
```

Comment:
If I’m not mistaken, [`timestamp`](https://wiki.openstreetmap.org/wiki/Elements#Common_attributes) is the timestamp of the changeset that last modified the way, so this heuristic could very well keep the way with _less_ history.

https://github.com/openstreetmap/iD/blob/1ee45ee1f03f0fe4d452012c65ac6ff7649e229f/modules/services/osm.js#L215

The way’s ID is a more reliable indicator of its age, assuming the way hasn’t ever been repurposed for something completely unrelated, so choose the way with the lowest nonnegative ID.

Reply:
good point, fixed